core/trino-main/src/main/java/io/trino/sql/planner/optimizations/PropertyDerivations.java

```diff
@@ -562,6 +564,14 @@ public ActualProperties visitExchange(ExchangeNode node, List<ActualProperties>
             // are computed for local exchanges.
             // TODO: implement full properties for local exchanges
             if (node.getScope() == LOCAL) {
+                if (inputProperties.size() == 1) {
+                    ActualProperties inputProperty = inputProperties.get(0);
+                    if (inputProperty.isEffectivelySingleStream() && node.getOrderingScheme().isEmpty()) {
```

Comment:
I'm still uncertain about the case of `node` being a repartition and `inputProperty` being grouped. @sopel39 is it safe to pass underlying GroupingProperty in such case?

Reply:
> @sopel39 is it safe to pass underlying GroupingProperty in such case?

Order of rows is preserved, so grouped rows should still be grouped in split streams.